src/components/LinksListInterface.vue

```diff
@@ -0,0 +1,112 @@
+<template>
+  <div
+    :title="'List for: ' + doi"
+  >
+    <div
+      v-if="(doi)"
+    >
+      <div v-if="typeof dataInput == 'undefined'">
+        <LinksList
+          :doi="doi"
+          :page="page"
+          :client-name="clientName"
+          :type="type"
+          :count="count"
+          @relationsLoaded="passCounter"
+          @referencesLoaded="passCounter"
+          @citationsLoaded="passCounter"
+        />
+      </div>
+      <div v-if="typeof dataInput !== 'undefined'">
+        <LinksListLocal
+          :doi="doi"
+          :page="page"
+          :client-name="clientName"
+          :type="type"
+          :data-input="dataObject"
+          @relationsLoaded="passCounter"
+          @referencesLoaded="passCounter"
+          @citationsLoaded="passCounter"
+        />
+      </div>
+    </div>
+    <a v-else>There is no DOI</a>
+  </div>
+</template>
+
+<script>
+import LinksListLocal from './LinksListLocal.vue';
+import LinksList from './LinksList.vue';
+import { utilsMixin } from '@/mixins/utilsMixin.js';
+
+export default {
+  name: 'LinksListInterface',
+  funtional: true,
+  components: {
+    LinksListLocal,
+    LinksList,
+  },
+  mixins: [utilsMixin],
+  props: {
+    // eslint-disable-next-line vue/require-default-prop
+    dataInput: {
+      type: String,
+    },
+    doi: {
+      type: String,
+      default: '',
+    },
+    type: {
+      type: String,
+      default: '',
+    },
+    clientName: {
+      type: String,
+      default: 'Datacite Search',
+    },
+    // eslint-disable-next-line vue/require-default-prop
+    page: {
+      type: String,
+      default: '1',
+    },
+    // eslint-disable-next-line vue/require-default-prop
+    count: {
+      type: Number,
+    },
+  },
+  computed: {
+    dataObject() {
+      if (typeof this.dataInput !== 'undefined') {
+        return JSON.parse(this.dataInput);
```

Comment:
Or here maybe if dataInput is always a string, you can then just do the parse and not worry about it.